parquet-hadoop/src/main/java/org/apache/parquet/hadoop/InternalParquetRecordWriter.java

```diff
@@ -154,6 +157,7 @@ private void checkBlockSizeReached() throws IOException {
 
   private void flushRowGroupToStore()
       throws IOException {
+    recordConsumer.flush();
```

Comment:
so this flushes a row group, what happens cross-page though? when a page gets compressed and stored in a big byte buffer, that page should contain the trailing nulls right?